main.py

```
@@ -287,17 +287,34 @@ def create_application() -> FastAPI:
+        allow_origins=cors_origins or ["*"],  # Fallback to wildcard if empty
         allow_credentials=True,
-        allow_methods=["*"],
+        allow_methods=["GET", "POST", "PUT", "DELETE", "PATCH", "OPTIONS"],
         allow_headers=["*"],
         expose_headers=["*"],
         max_age=86400  # Cache preflight for 24 hours
     )
     
+    # Add explicit OPTIONS handler for troubleshooting
+    @app.options("/{path:path}")
+    async def handle_options(path: str):
```

Ruff (0.12.2)

Unused function argument: `path`

(ARG001)